### DISCUSSION TITLE
make code navigation available via url

### DISCUSSION THREAD

**Discussion by MrNom4ik** [community/community#60948] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

As you know, you can specify a specific line in your code with the `#L<line number>` fragment at the end of the url.
Now github uses this to point to certain classes/functions in the code (I'm talking about the window on the right in which there is the whole code structure)
But it seems to me that this has a problem. For example, if a user in a markdown documentation file has a link to a certain class/function in the source code, and then he changes this file, then this link will point to another line, not the one on which the desired class/function is located.
Why not do something similar to `#<object>`? for example:
`https://github.com/python/cpython/blob/main/Lib/cProfile.py#L46`
will change to
`https://github.com/python/cpython/blob/main/Lib/cProfile.py#Profile.dump_stats`

**Comment by Relisfoku** [comment]
https://github.com/python/cpython/blob/main/Lib/cProfile.py#L46

**Comment by look** [comment]
Thanks for the suggestion @MrNom4ik! That is a cool idea, I've passed it along to the frontend team.

To accomplish something similar, I've started linking to searches rather than files. For example, you could link to a search for [`repo:python/cpython path:Lib/cProfile.py def:dump_stats`](https://github.com/search?q=repo%3Apython%2Fcpython+path%3ALib%2FcProfile.py+def%3Adump_stats&type=code). This will always point to the most recent version. You can also loosen the search (for example, don't limit it to the path) to guard against code being moved to a different file.
